### PR TITLE
Update changelog

### PR DESCRIPTION
**Pull request by henryiii**
Updating the changelog, and moving the single-file generation to Github Actions. As a result, you can now download the current single file header from any Github Actions build via an artifact. Click details -> download artifact (upper righthand corner).

### CODE REVIEW

**Comment by henryiii** [comment]
@phlptp Can you go over the changelog and make sure it is clear and representative of the changes made?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,13 +1,44 @@
-## Version 1.9: IN PROGRESS
+## Version 1.9: Config files and cleanup
 
-* The meson build system supported [#299][]
+Config file handling was revamped to fix common issues, and now supports TOML.
```
Comment (phlptp) [inline CHANGELOG.md]:
```suggestion
Config file handling was revamped to fix common issues, and now supports reading [TOML](https://github.com/toml-lang/toml).

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,13 +1,44 @@
...
+configuration options were added to facilitate a wider variety of apps.  GCC
+4.7 is no longer supported.
+
+* Config files refactored, supports TOML (may become default output in 2.0) [#362][]
 * Added two template parameter form of `add_option`, allowing `std::optional` to be supported without a special import [#285][]
 * `string_view` now supported in reasonable places [#300][], [#285][]
 * `app.immediate_callback()` allows the main app to run before subcommand callbacks. [#292][]
-* GCC 4.7 is no longer supported, due mostly to GoogleTest. GCC 4.8+ is now required. [#160][]
+* Multiple positional arguments maintain order if `positionals_at_end` is set. [#306][]
+* Tuple/array now supported [#307][], [#310][]
```
Comment (phlptp) [inline CHANGELOG.md]:
```suggestion
* Pair/tuple/array now supported, and validators indexed to specific components in the objects [#307][], [#310][]
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,13 +1,44 @@
...
+Config file handling was revamped to fix common issues, and now supports TOML.
+Adding options is significantly more powerful with support for things like
+`std::tuple` and `std::array`, including with transforms. Several new
+configuration options were added to facilitate a wider variety of apps.  GCC
+4.7 is no longer supported.
+
+* Config files refactored, supports TOML (may become default output in 2.0) [#362][]
 * Added two template parameter form of `add_option`, allowing `std::optional` to be supported without a special import [#285][]
 * `string_view` now supported in reasonable places [#300][], [#285][]
 * `app.immediate_callback()` allows the main app to run before subcommand callbacks. [#292][]
```
Comment (phlptp) [inline CHANGELOG.md]: this is closely related to #313 so we may want to group those together.

**Comment by phlptp** [comment]
need to add a note about #387  assuming that gets merged

**Review by phlptp — APPROVED** [review]
looks good to me.

**Comment by henryiii** [comment]
Will probably bump the version and release early next week if no issues are found. Docs updates/bugfixes welcome until then. :) New code will cause a delay (which is not bad if it's worth it, though).

**Pull request merged**